kolibri_instant_schools_plugin/management/commands/migratev1database.py

```diff
@@ -0,0 +1,188 @@
+from kolibri.auth.models import Facility, FacilityUser, Role
+from kolibri.core.device.models import DevicePermissions, DeviceSettings
+from kolibri.logger.models import ContentSummaryLog, ContentSessionLog, MasteryLog, UserSessionLog, AttemptLog
+from ...models import PhoneToUsernameMapping
+
+from sqlalchemy import MetaData, create_engine
+from sqlalchemy.ext.automap import automap_base
+from sqlalchemy.orm import sessionmaker
+
+USER_ID_MAPPING = {}
+SUMMARYLOG_ID_MAPPING = {}
+MASTERYLOG_ID_MAPPING = {}
+SESSIONLOG_ID_MAPPING = {}
+
+class Command(BaseCommand):
+    help = "Migrate data from the old Instant Schools v1 database into the v2 database."
+
+    requires_system_checks = False
+
+    def handle(self, *args, **options):
+
+        # ensure the new databases have both been migrated
+        call_command("migrate")
+        call_command("migrate", database="instant_schools")
+        
+        # load the old databases
+        old_db_session, old_classes = self.load_database(settings.OLD_DB_PATH)
+        old_hash_db_session, old_hash_classes = self.load_database(settings.OLD_HASH_DB_PATH)
+
+        # load the SQLAlchemy ORM classes for the old databases
+        Collection_ = old_classes.kolibriauth_collection
+        DeviceOwner_ = old_classes.kolibriauth_deviceowner
+        FacilityUser_ = old_classes.kolibriauth_facilityuser
+        ContentSummaryLog_ = old_classes.logger_contentsummarylog
+        ContentSessionLog_ = old_classes.logger_contentsessionlog
+        AttemptLog_ = old_classes.logger_attemptlog
+        MasteryLog_ = old_classes.logger_masterylog
+        UserSessionLog_ = old_classes.logger_usersessionlog
+        Lookup_ = old_hash_classes.lookup
+
+        # MIGRATE: Facility
+        old_facility = old_db_session.query(Collection_).filter(Collection_.kind == "facility")[0]
+        new_facility, _ = Facility.objects.get_or_create(name=old_facility.name)
+        new_facility.dataset.preset = "nonformal"
+        new_facility.dataset.learner_can_edit_username = False
+        new_facility.dataset.save()
+
+        # MIGRATE: DeviceOwner
+        for deviceowner in old_db_session.query(DeviceOwner_).all():
+            superuser, _ = FacilityUser.objects.get_or_create(
+                username=deviceowner.username,
+                full_name=deviceowner.full_name,
+                password=deviceowner.password,
+                date_joined=deviceowner.date_joined,
+                last_login=deviceowner.last_login,
+                facility=new_facility,
+            )
+            Role.objects.get_or_create(user=superuser, collection=new_facility, kind=ADMIN)
+            DevicePermissions.objects.get_or_create(user=superuser, is_superuser=True)
+
+        # MIGRATE: FacilityUser
+        for old_user in old_db_session.query(FacilityUser_).all():
+
+            # migrate the phone number mapping
+            old_username = old_user.username
+            if len(old_username) == 30:
```

Comment:
As we limited the phonenumbers themselves to 10 digits (?) this should never be the case for an unhashed phone number, I assume?

Reply:
Yeah, phone numbers should all be max... at most 11, I think? I believe our validation of this is primarily client-side, but I've skimmed through the data this will be affecting and all the phone numbers are short. Certainly shorter than 30, in any case! :)